File: backend/src/main/java/br/com/sfpacim/backend/dtos/erro/CampoMensagemDTO.java
package br.com.sfpacim.backend.dtos.erro;

import io.swagger.v3.oas.annotations.media.Schema;

/**
 * Detalha um erro de validação específico em um campo.
 *
 * @author Matheus F. N. Pereira
 */
@Schema(description = "Detalha um erro de validação específico em um campo.")
public record CampoMensagemDTO(

        @Schema(description = "Nome do campo onde ocorreu o erro.", example = "email") //
        String campo,

        @Schema(description = "Mensagem de erro associada ao campo.", example = "Campo obrigatório") //
        String mensagem) {
}

File: frontend/src/app/dtos/erro/CampoMensagemDTO.ts
/**
 * DTO (Interface) que detalha um erro de validação específico
 * em um único campo do formulário.
 *
 * @author Matheus F. N. Pereira
 */
export interface CampoMensagemDTO {
  /**
   * O nome do campo que falhou na validação (ex: "senha").
   */
  campo: string;

  /**
   * A mensagem de erro específica para aquele campo (ex: "Este campo é obrigatório.").
   */
  mensagem: string;
}
